app/config/taxRatesAndBands.scala

```diff
@@ -47,15 +48,27 @@ object TaxRatesAndBands {
 
   val allRates: List[TaxRatesAndBands] = TaxRatesAndBands20152016 :: TaxRatesAndBands20162017 :: TaxRatesAndBands20172018 ::
     TaxRatesAndBands20182019 :: TaxRatesAndBands20192020 :: TaxRatesAndBands20202021 :: TaxRatesAndBands20212022 ::
-    TaxRatesAndBands20222023 :: TaxRatesAndBands20232024 :: TaxRatesAndBands20242025 :: Nil
+    TaxRatesAndBands20222023 :: TaxRatesAndBands20232024 :: TaxRatesAndBands20242025 :: TaxRatesAndBands20242025MidYearChange :: Nil
 
   val liveTaxRates: List[TaxRatesAndBands] = if (LocalDate.now.isBefore(latestTaxYearGoLiveDate)) allRates.dropRight(1) else allRates
 
-  def getRates(year: Int): TaxRatesAndBands = liveTaxRates.filter(_.taxYear == year) match {
-    case params if params.nonEmpty => params.head
+  def getRates(year: Int, disposalDate : Option[LocalDate] = None , isMidYearChangeApplicable : Boolean = false): TaxRatesAndBands = liveTaxRates.filter(_.taxYear == year) match {
+    case params if params.nonEmpty => getTaxRatesAndBandsForMidYearChange(params, disposalDate,isMidYearChangeApplicable)
     case _ => liveTaxRates.maxBy(_.taxYear)
```

Comment:
do we need to be concerned with what this case is doing?

Reply:
moved the if statement.

Reply:
No, that's the default case as is in current logic and in case of not finding a valid tax year config we pick the max of configs available. That's current behaviour aswell

Reply:
Yep, that's the case have just pushed up as part of simplifying the condition blocks